nemo_curator/modules/semantic_dedup.py

```diff
@@ -216,13 +225,22 @@ def __call__(self, dataset: DocumentDataset) -> DocumentDataset:
                     embedding_ddf,
                     self.embedding_output_dir,
                     write_to_filename=self.write_to_filename,
-                    output_type="parquet",
+                    output_type=self.output_file_type,
+                )
+
+            if self.output_file_type == "jsonl":
+                embedding_files = get_all_files_paths_under(self.embedding_output_dir)
+                ddf = DocumentDataset(
+                    dask_cudf.read_json(embedding_files, blocksize="2GB")
```

Comment:
We should probably move to loading from `DocumentDataset.from_parquet` , also lets to write stuff to `jsonl` for intermediate structures